bin/inx-chronicle/src/api/metrics.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{pin::Pin, task::Poll, time::SystemTime};
+
+use axum::{routing::get, Router};
+use futures::Future;
+use hyper::{Method, Request, Response, Uri};
+use lazy_static::lazy_static;
+use prometheus::{Encoder, Gauge, HistogramOpts, HistogramVec, IntCounter, IntCounterVec, Opts, Registry, TextEncoder};
+use tower::{Layer, Service};
+
+use super::error::APIError;
+
+pub fn routes() -> Router {
+    Router::new().route("/metrics", get(metrics))
+}
+
+async fn metrics() -> Result<String, APIError> {
+    let encoder = TextEncoder::new();
+    let mut buffer = Vec::new();
+    encoder
+        .encode(&REGISTRY.gather(), &mut buffer)
+        .map_err(APIError::other)?;
+
+    encoder
+        .encode(&prometheus::gather(), &mut buffer)
+        .map_err(APIError::other)?;
+
+    String::from_utf8(buffer).map_err(APIError::other)
+}
+
+pub fn register_metrics() {
+    REGISTRY
+        .register(Box::new(INCOMING_REQUESTS.clone()))
+        .expect("Could not register collector");
+
+    REGISTRY
+        .register(Box::new(RESPONSE_CODE_COLLECTOR.clone()))
+        .expect("Could not register collector");
+
+    REGISTRY
+        .register(Box::new(RESPONSE_TIME_COLLECTOR.clone()))
+        .expect("Could not register collector");
+
+    REGISTRY
+        .register(Box::new(CONFIRMATION_TIME_COLLECTOR.clone()))
+        .expect("Could not register collector");
+}
+
+lazy_static! {
```

Comment:
@grtlr 

> Would there be a way to implement this without using lazy_static? In bee-metrics we did something similar: https://github.com/iotaledger/bee/commit/8efaa02efe076da88a7d23ff1d1ace0aabe5c12a

Reply:
The `once_cell` crate allows you to have lazy initialized global data without using a macro.